Sources/SwiftSyntaxMacroExpansion/MacroExpansion.swift

```diff
@@ -38,18 +40,23 @@ extension MacroRole {
     case .conformance: return "ConformanceMacro"
     case .codeItem: return "CodeItemMacro"
     case .extension: return "ExtensionMacro"
+    case .preamble: return "PreambleMacro"
+    case .body: return "BodyMacro"
     }
   }
 }
 
 /// Simple diagnostic message
-private enum MacroExpansionError: Error, CustomStringConvertible {
+enum MacroExpansionError: Error, CustomStringConvertible {
   case unmatchedMacroRole(Macro.Type, MacroRole)
   case parentDeclGroupNil
   case declarationNotDeclGroup
   case declarationNotIdentified
+  case declarationHasNoBody
   case noExtendedTypeSyntax
   case noFreestandingMacroRoles(Macro.Type)
+  case tooManyBodyMacros
```

Comment:
I would prefer `moreThanOneBodyMacro`. `tooManyBodyMacros` makes me wonder: How many is too many? 2, 3, 20?